streams/src/main/java/org/apache/kafka/streams/kstream/Repartitioned.java

```diff
@@ -0,0 +1,177 @@
+                          final Serde<V> valueSerde,
+                          final Integer numberOfPartitions,
+                          final StreamPartitioner<K, V> partitioner) {
+        this.name = name;
+        this.keySerde = keySerde;
+        this.valueSerde = valueSerde;
+        this.numberOfPartitions = numberOfPartitions;
+        this.partitioner = partitioner;
+    }
+
+    protected Repartitioned(final Repartitioned<K, V> repartitioned) {
+        this(
+            repartitioned.name,
+            repartitioned.keySerde,
+            repartitioned.valueSerde,
+            repartitioned.numberOfPartitions,
+            repartitioned.partitioner
+        );
+    }
+
+    /**
+     * Create a {@code Repartitioned} instance with the provided name used as part of the repartition topic.
+     *
+     * @param name the name used as a processor named and part of the repartition topic name.
+     * @param <K>  key type
+     * @param <V>  value type
+     * @return A new {@code Repartitioned} instance configured with processor name and repartition topic name
+     * @see KStream#repartition(Repartitioned)
+     * @see KStream#repartition(KeyValueMapper, Repartitioned)
+     */
+    public static <K, V> Repartitioned<K, V> as(final String name) {
+        return new Repartitioned<>(name, null, null, null, null);
+    }
+
+    /**
+     * Create a {@code Repartitioned} instance with provided key serde and value serde.
+     *
+     * @param keySerde   Serde to use for serializing the key
+     * @param valueSerde Serde to use for serializing the value
+     * @param <K>        key type
+     * @param <V>        value type
+     * @return A new {@code Repartitioned} instance configured with key serde and value serde
+     * @see KStream#repartition(Repartitioned)
+     * @see KStream#repartition(KeyValueMapper, Repartitioned)
+     */
+    public static <K, V> Repartitioned<K, V> with(final Serde<K> keySerde,
+                                                  final Serde<V> valueSerde) {
+        return new Repartitioned<>(null, keySerde, valueSerde, null, null);
+    }
+
+    /**
+     * Create a {@code Repartitioned} instance with provided partitioner.
+     *
+     * @param partitioner the function used to determine how records are distributed among partitions of the topic,
+     *                    if not specified and the key serde provides a {@link WindowedSerializer} for the key
+     *                    {@link WindowedStreamPartitioner} will be used—otherwise {@link DefaultPartitioner} will be used
+     * @param <K>         key type
+     * @param <V>         value type
+     * @return A new {@code Repartitioned} instance configured with partitioner
+     * @see KStream#repartition(Repartitioned)
+     * @see KStream#repartition(KeyValueMapper, Repartitioned)
+     */
+    public static <K, V> Repartitioned<K, V> streamPartitioner(final StreamPartitioner<K, V> partitioner) {
```

Comment:
It seems this method is unused. Are we missing test coverage?

Reply:
Added tests.